lib/libc/Kconfig

```diff
@@ -40,7 +40,8 @@ endchoice # LIBC_IMPLEMENTATION
 
 if NEWLIB_LIBC
 
-if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb"
+if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb" || \
+  ("$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr" && "${SDK_MAJOR_MINOR}" = "0.11")
```

Comment:
Adding this for each SDK version is not going to scale.

Reply:
I believe it was explained to me that libc-nano was the only supported newlib variant for zephyr-sdk prior to 0.11.

So shouldn't we omit this version check?